server/sonar-server/src/main/java/org/sonar/server/plugins/ServerPluginRepository.java

```diff
@@ -316,10 +316,16 @@ public void uninstall(String pluginKey, File uninstallDir) {
 
     for (String uninstallKey : uninstallKeys) {
       PluginInfo info = getPluginInfo(uninstallKey);
+
       try {
+        if (!getPluginFile(info).exists()) {
```

Comment:
you should call `getPluginFile(info)` only once (rather than here and at line 328). Use a variable

Reply:
If I call api/plugins/uninstall with plugin key foo, I'll get message "Plugin already uninstalled" while it's not true, there is no such plugin, at all.

To be more reliable, if `getPluginFile(info)` we should test file exists in uninstallDir. If so, display this message, if not, throw an error.

Reply:
The plugin info is loaded from memory. So the plugin existed at startup and was removed. 
I will use a var for the plugin file (even though this does not involve I/O)